src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4115,23 +4115,26 @@ private ParameterSyntax ParseParameter()
 
                 TypeSyntax type;
                 SyntaxToken name;
-                SyntaxToken exclamation = null;
+                SyntaxToken exclamationExclamation = null;
                 SyntaxToken equals = null;
                 if (this.CurrentToken.Kind != SyntaxKind.ArgListKeyword)
                 {
                     type = this.ParseType(mode: ParseTypeMode.Parameter);
                     name = this.ParseIdentifierToken();
                     if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
                     {
-                        exclamation = this.EatToken(SyntaxKind.ExclamationToken);
-                    }
-
-                    else if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken)
-                    {
-                        var notEq = this.EatToken(SyntaxKind.ExclamationEqualsToken);
-                        equals = ConvertToMissingWithTrailingTrivia(notEq, SyntaxKind.EqualsToken);
-                        equals = AddError(equals, ErrorCode.ERR_NeedSpaceBetweenExclamationAndEquals);
-                        exclamation = SyntaxFactory.MissingToken(SyntaxKind.ExclamationToken);
+                        exclamationExclamation = this.EatToken();
+                        if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken)
+                        {
+                            var exclamationEquals = this.EatToken();
+                            var exclamation2 = SyntaxFactory.Token(exclamationEquals.GetLeadingTrivia(), SyntaxKind.ExclamationToken, null);
```

Comment:
>exclamation2 [](start = 32, length = 12)

Consider renaming to `exclamation` since there's no `exclamation[1]`.

Reply:
Might do that, or rename it to secondExclamation to emphasize that it's meant to be a part of the other Exclamation.